src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/complex-navigator-utils.ts

```diff
@@ -0,0 +1,47 @@
+import {Element} from '../../../../shared/model/complex-results/element.model';
+import {ComplexComponent} from '../../../../shared/model/complex-results/complex-component.model';
+
+export function formatStoichiometryValues(stoichiometry: string): string {
+  if (!!stoichiometry) {
+    const matchedStoichiometry = fetchValuesFromStoichiometry(stoichiometry);
+    if (!!matchedStoichiometry) {
+      const minValue = parseInt(matchedStoichiometry[1], 10);
+      const maxValue = parseInt(matchedStoichiometry[2], 10);
+      if (minValue === maxValue) {
+        return minValue.toString();
+      } else {
+        return `${minValue}, ${maxValue}`;
+      }
+    }
+  }
+  return ' '; // sometimes we don't have the stoichiometry value
+}
+
+export function fetchValuesFromStoichiometry(stoichiometry: string) {
```

Comment:
Right now this function is used in multiple places with similar code. We can make some changes to reduce code duplication.

First, we have the method `formatStoichiometryValues` in this same class.
Then, the method `addedStoichiometryValues` in table-main-interactor.component.ts
These 2 methods are quite similar and we can reduce code duplication by doing.

In this class, add a new method
```
export function parseStoichiometryValues(stoichiometry: string): [number, number] {
  if (!!stoichiometry) {
    const matchedStoichiometry = fetchValuesFromStoichiometry(stoichiometry);
    if (!!matchedStoichiometry) {
      const minValue = parseInt(matchedStoichiometry[1], 10);
      const maxValue = parseInt(matchedStoichiometry[2], 10);
      if (!isNaN(minValue) && !isNaN(maxValue)) {
        return [minValue, maxValue];
      }
    }
  }
  return null;
}
```

And also in this class update `formatStoichiometryValues`
```
export function formatStoichiometryValues(stoichiometry: string): string {
  const parsedStoichiometry = parseStoichiometryValues(stoichiometry);
  if (!!parsedStoichiometry) {
    const minValue = parsedStoichiometry[0];
    const maxValue = parsedStoichiometry[1];
    if (minValue === maxValue) {
      return minValue.toString();
    } else {
      return `${minValue}, ${maxValue}`;
    }
  }
  return ' '; // sometimes we don't have the stoichiometry value
}
```

Finally, in table-main-interactor.component.ts, update `addedStoichiometryValues` to
```
  private addedStoichiometryValues(components: ComplexComponent[]): [number, number] {
    let minValue: number = null;
    let maxValue: number = null;
    for (const component of components) {
      const parsedStoichiometry = parseStoichiometryValues(component.stochiometry);
      if (!!parsedStoichiometry) {
        if (minValue === null) {
          minValue = 0;
        }
        if (maxValue === null) {
          maxValue = 0;
        }
        minValue += parsedStoichiometry[0];
        maxValue += parsedStoichiometry[1];
      }
    }
    if (minValue !== null && maxValue !== null) {
      return [minValue, maxValue];
    }
    return null;
  }
```